src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,222 @@
+export chow_ring, augmented_chow_ring, select
+
+@doc Markdown.doc"""
+    chow_ring(M::Matroid, ring::MPolyRing=nothing; extended=false)
+
+Return the Chow ring of a matroid, optional also with the simplicial generators.
+
+See [AHK18](@cite) and [BES21](@cite). 
+
+# Examples
+The following computes the chow ring of the Fano matroid.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M);
+
+julia> R[1]*R[8]
+-x_{3,4,7}^2
+```
+
+# Examples
+The following computes the chow ring of the Fano matroid including variables for the simplicial generators.
+```jldoctest
+julia> M = fano_matroid();
+
+julia> R = chow_ring(M, extended=true);
+
+julia> f = R[22] + R[8] - R[29]
+x_{1,2,3} + h_{1,2,3} - h_{1,2,3,4,5,6,7}
+
+julia> f==0
+true
+```
+
+# Examples
+The following computes the chow ring of the Fano matroid including variables for the simplicial generators.
+```jldoctest
+julia> M = uniform_matroid(3,3);
+
+julia> GR, _ = GradedPolynomialRing(QQ,["a","b","c","d","e","f"]);
+
+julia> R = chow_ring(M, GR);
+
+julia> hilbert_series_reduced(R)
+(t^2 + 4*t + 1, 1) 
+
+```
+"""
+function chow_ring(M::Matroid, ring::Union{MPolyRing,Nothing}=nothing; extended::Bool=false)
+        is_loopless(M) || error("Matroid has loops")
+	Flats = flats(M)
```

Comment:
```suggestion
    Flats = flats(M)
```
Please don't use tabs.

And *never* mix tabs and spaces as in this code here.